fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+							   "Unspoiled","Disgraced","Defrocked","Shamed","Meek","Timid","Broken")//,"Fresh")
+
+/datum/antagonist/bloodsucker/proc/AmFledgling()
+	return !bloodsucker_title
+
+/datum/antagonist/bloodsucker/proc/ReturnFullName(var/include_rep=0)
+	var/fullname
+	// Name First
+	fullname = (bloodsucker_name ? bloodsucker_name : owner.current.name)
+	// Title
+	if(bloodsucker_title)
+		fullname = bloodsucker_title + " " + fullname
+	// Rep
+	if(include_rep && bloodsucker_reputation)
+		fullname = fullname + " the " + bloodsucker_reputation
+	return fullname
+
+//Bloodsucker team
+/datum/team/bloodsucker
+	name = "Bloodsucker"
+
+/datum/antagonist/bloodsucker/get_team()
+	return bloodsucker_team
+
+/datum/antagonist/bloodsucker/create_team(datum/team/bloodsucker/new_team)
+	if(!new_team)
+		for(var/datum/antagonist/bloodsucker/H in GLOB.antagonists)
+			if(!H.owner)
+				continue
+			if(H.bloodsucker_team)
+				bloodsucker_team = H.bloodsucker_team
+				return
+	if(!istype(new_team))
+		stack_trace("Wrong team type passed to [type] initialization.")
+	bloodsucker_team = new_team
+
+//Individual roundend report
+/datum/antagonist/bloodsucker/roundend_report()
+	var/list/report = list()
+
+	// Vamp Name
+	report += "<br><span class='header'><b>\[[ReturnFullName(TRUE)]\]</b></span>"
+
+	// Default Report
+	report += ..()
+
+	// Now list their vassals
+	if (vassals.len > 0)
+		report += "<span class='header'>Their Vassals were...</span>"
+		for (var/datum/antagonist/vassal/V in vassals)
+			if (V.owner)
+				var/jobname = V.owner.assigned_role ? "the [V.owner.assigned_role]" : ""
+				report += "<b>[V.owner.name]</b> [jobname]"
+
+	return report.Join("<br>")
+
+//Called when using admin tools to give antag status
+/datum/antagonist/bloodsucker/admin_add(datum/mind/new_owner,mob/admin)
+	message_admins("[key_name_admin(admin)] made [key_name_admin(new_owner)] into [name].")
+	log_admin("[key_name(admin)] made [key_name(new_owner)] into [name].")
+	new_owner.add_antag_datum(src)
+
+//Called when removing antagonist using admin tools
+/datum/antagonist/bloodsucker/admin_remove(mob/user)
+	if(!user)
+		return
+	message_admins("[key_name_admin(user)] has removed [name] antagonist status from [key_name_admin(owner)].")
+	log_admin("[key_name(user)] has removed [name] antagonist status from [key_name(owner)].")
+	on_removal()
+
+// Buying powers
+/datum/antagonist/bloodsucker/proc/BuyPower(datum/action/bloodsucker/power)//(obj/effect/proc_holder/spell/power)
+	powers += power
+	power.Grant(owner.current) // owner.AddSpell(power)
+
+/datum/antagonist/bloodsucker/proc/AssignStarterPowersAndStats()
+	update_hud(owner.current) 	// Set blood value, current rank
+	// Powers
+	BuyPower(new /datum/action/bloodsucker/feed)
+	BuyPower(new /datum/action/bloodsucker/masquerade)
+	BuyPower(new /datum/action/bloodsucker/veil)
+	// Traits
+	if(HAS_TRAIT(owner.current, TRAIT_TOXINLOVER)) //No slime bonuses here, no thank you
```

Comment:
what if you receive TRAIT_TOXINLOVER by turning into a slime later?

why not just make TRAIT_TOXINLOVER not regenerate the blood of vampires when they take tox damage instead of entirely removing it from them? A bloodsucker who doesn't know about this feature could easily chug a shitton of toxins as a slimeperson and die in an incredibly lame way.